pkg/controllers/management/kontainerdrivermetadata/data_getter.go

```diff
@@ -0,0 +1,181 @@
+		}
+		if addon.Labels[sendRKELabel] == "false" {
+			logrus.Infof("sendRKELabel false addonName %s, k8sVersion %s", addonName, k8sVersion)
+			return "", nil
+		}
+		return addon.Template, err
+	}
+
+	for _, name := range names {
+		addon, err := addons.GetNamespaced(namespace.GlobalNamespace, name, metav1.GetOptions{})
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return "", err
+			}
+			continue
+		}
+		if addon.Labels[sendRKELabel] == "false" {
+			logrus.Infof("sendRKELabel false addonName %s, k8sVersion %s", addonName, k8sVersion)
+			return "", nil
+		}
+		return addon.Template, err
+	}
+	return "", nil
+}
+
+func GetRKEK8sServiceOptions(k8sVersion string, svcOptionLister v3.RKEK8sServiceOptionLister, svcOptions v3.RKEK8sServiceOptionInterface) (*v3.KubernetesServicesOptions, error) {
+	names := []string{
+		getName(k8sVersion),
+		getName(util.GetTagMajorVersion(k8sVersion)),
+	}
+	var k8sSvcOption *v3.KubernetesServicesOptions
+	for _, name := range names {
+		obj, err := svcOptionLister.Get(namespace.GlobalNamespace, name)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return k8sSvcOption, err
+			}
+			continue
+		}
+		if obj.Labels[sendRKELabel] == "false" {
+			logrus.Infof("svcOption false k8sVersion %s", k8sVersion)
+			return k8sSvcOption, nil
+		}
+		return &obj.ServiceOptions, nil
+	}
+
+	for _, name := range names {
+		obj, err := svcOptions.GetNamespaced(namespace.GlobalNamespace, name, metav1.GetOptions{})
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return k8sSvcOption, err
+			}
+			continue
+		}
+		if obj.Labels[sendRKELabel] == "false" {
+			logrus.Infof("svcOption false k8sVersion %s", k8sVersion)
+			return k8sSvcOption, nil
+		}
+		return &obj.ServiceOptions, nil
+	}
+	return k8sSvcOption, nil
+}
+
+func GetK8sVersionInfo(rancherVersion string, rkeSysImages map[string]v3.RKESystemImages,
+	winSysImages map[string]v3.WindowsSystemImages, svcOptions map[string]v3.KubernetesServicesOptions,
+	rancherVersions map[string]v3.K8sVersionInfo) (map[string]v3.RKESystemImages, map[string]v3.WindowsSystemImages, map[string]v3.KubernetesServicesOptions) {
+
+	k8sVersionRKESystemImages := map[string]v3.RKESystemImages{}
+	k8sVersionWinSystemImages := map[string]v3.WindowsSystemImages{}
+	k8sVersionSvcOptions := map[string]v3.KubernetesServicesOptions{}
+
+	//rancherVersion := getRancherVersion()
+	maxVersionForMajorK8sVersion := map[string]string{}
+	for k8sVersion := range rkeSysImages {
+		if rancherVersionInfo, ok := rancherVersions[k8sVersion]; ok {
+			greaterThanMax := rancherVersionInfo.MaxRancherVersion != "" && rancherVersion > rancherVersionInfo.MaxRancherVersion
+			lowerThanMin := rancherVersionInfo.MinRancherVersion != "" && rancherVersion < rancherVersionInfo.MinRancherVersion
+			if greaterThanMax || lowerThanMin {
+				continue
+			}
+		}
+		majorVersion := util.GetTagMajorVersion(k8sVersion)
+		if curr, ok := maxVersionForMajorK8sVersion[majorVersion]; !ok || k8sVersion > curr {
```

Comment:
semver comparison would be ideal

Reply:
updating to use semver for comparing rancher version info, but using string comparison for finding max since it doesn't respect semantic versioning

checked in 1.6, seems like we did something similar there too 
```
                long currentMaxVersion = Long.valueOf(url.substring(url.lastIndexOf(":") + 1, url.length()));
                if (currentMaxVersion >= maxVersion) {
                    maxVersion = currentMaxVersion;
```